bundles/statistics/statsgrid/FlyoutManager.js

```diff
@@ -91,10 +87,17 @@ Oskari.clazz.define('Oskari.statistics.statsgrid.FlyoutManager', function (insta
         if (type === 'search') {
             this.searchHandler.getController().toggleSearchFlyout(true, () => this.trigger('hide', type));
             this.hide('table');
+            this.hide('diagram');
             this.trigger('show', type);
         } else if (type === 'table') {
             this.tableHandler.getController().toggleTableFlyout(true, () => this.trigger('hide', type));
             this.hide('search');
+            this.hide('diagram');
+            this.trigger('show', type);
+        } else if (type === 'diagram') {
+            this.diagramHandler.getController().toggleDiagramFlyout(true, () => this.trigger('hide', type));
+            this.hide('search');
+            this.hide('table');
             this.trigger('show', type);
         } else {
```

Comment:
Now that these are all React-based, could we do somehing like:
```
 this.flyoutInfo[type].controller.toggleFlyout(true, () => this.trigger('hide', type));
 Object.keys(this.flyoutInfo)
    .filter(typeId => typeId !== type)
    .forEach(typeId =>  this.hide(typeId)) ;
```

This assumes that this.flyoutInfo is changed from array to an object with the type/id of the previous object as key.